### ISSUE TITLE
[web] implement platformViewId in the semantics

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#101439]
Flutter communicates the semantics of platform views by providing a semantics node with a [platformViewId](https://api.flutter.dev/flutter/semantics/SemanticsNode/platformViewId.html). It then expects the engine to embed the semantics of the platform view into Flutter's semantics tree for proper presentation to the user.

On the web, there is no way to separate the semantics of HTML from its visual representation, and so it's not possible to render a platform view's pixels in the render tree, and its semantics into a separate semantics tree.

Currently, on the web we render semantics either on top of the render tree or below it[^1]. This works in simple cases, but it doesn't cover everything. For example, a video player covered by a pointer interceptor is in a dilemma:

- If the semantics tree is on top, it obscured the semantics of the video player.
- If the semantics tree is on the bottom, then it is obscured by the pointer interceptor.

A more correct implementation would segment the semantics tree such that Flutter semantics that must appear behind a platform view are rendered behind it, and those on top are rendered on top. [Scenelet design](flutter.dev/go/flutter-web-scenelets) touches on how this can be done if we prerolled the scene and semantics into a sequence of "scenelets", but this probably needs a dedicated design doc.

[^1]: a developer-facing API this not yet available but is about to be implemented.

**Comment by tgrushka** [comment]
Yes, please Flutter team, address this if possible. As an example, when I embed a map (e.g. Google Map or MapBox map, whatever) in my Flutter app, it renders an `flt-platform-view`, which prepends its `Semantics` to the beginning of the web page. This cannot be overridden by wrapping the plugin widget (which renders the platform view) in `Semantics`, `ExcludeSemantics`, or anything.

I think the only way to override (but a very bad practice) is to add JavaScript to my index.html to listen for DOM changes, or inject from Dart, some JavaScript that will add `aria-hidden="true"` to all the `flt-platform-view` inside the `flt-glass-pane` shadow DOM. Would be really great to be able to just have the `Semantics` for these views show up in proper DOM order. Thank you.

**Comment by yjbanov** [comment]
Update: this is now partially addressed via `aria-owns` attribute. Unfortunately, Safari doesn't support it yet https://bugs.webkit.org/show_bug.cgi?id=223798, so we can't consider it fully fixed yet.
